context.js

```diff
@@ -140,9 +140,9 @@ class Context {
     return this
   }
 
-  unexpectedError (err) {
+  unexpectedError (err, formatted) {
     this.errors.push(err)
-    this.output = String((err && err.stack) || err)
+    this.output = formatted || String((err && err.stack) || err)
```

Comment:
Let's use the error formatter here, making sure it is a function before we use it:

```suggestion
    this.output = typeof this.errorFormatter === 'function'
      ? String(this.errorFormatter(err))
      : String((err && err.stack) || err)
```

and then accept the error formatter in the Context constructor, like so:

```js
this.errorFormatter = opts.errorFormatter
```